### ISSUE TITLE
VariantDataset globals property not working

### ISSUE THREAD

**Issue by jigold** [hail-is/hail#1368]
Issue came up on doctest branch.

Reproducible example:
```
assoc_vds = hc.import_vcf('src/test/resources/sample.vcf')
  .split_multi()
  .variant_qc()
  .annotate_samples_expr('sa.culprit = gs.filter(g => v == Variant("20", 13753124, "A", "C")).map(g => g.gt).collect()[0]')
  .annotate_samples_expr('sa.pheno = rnorm(1,1) * sa.culprit')
  .annotate_samples_expr('sa.cov1 = rnorm(0,1)')
  .annotate_samples_expr('sa.cov2 = rnorm(0,1)')
  .linreg('sa.pheno', ['sa.cov1', 'sa.cov2']).annotate_variants_expr('va.useInKinship = va.qc.AF > 0.05')

kinship_vds = assoc_vds.filter_variants_expr('va.useInKinship')
lmm_vds = assoc_vds.lmmreg(kinship_vds, 'sa.pheno', ['sa.cov1', 'sa.cov2'])

lmm_vds.globals
```

Error message:
```
Failed example:
    lmm_vds.globals
Exception raised:
    Traceback (most recent call last):
      File "//anaconda/lib/python2.7/doctest.py", line 1315, in __run
        compileflags, 1) in test.globs
      File "<doctest default[1]>", line 1, in <module>
        lmm_vds.globals
      File "/Users/jigold/hail/python/hail/dataset.py", line 1958, in globals
        self._globals = self.global_schema._convert_to_py(self._jvds.globalAnnotation())
      File "/Users/jigold/hail/python/hail/type.py", line 423, in _convert_to_py
        d[f.name] = f.typ._convert_to_py(annotation.get(i))
      File "/Users/jigold/hail/python/hail/type.py", line 423, in _convert_to_py
        d[f.name] = f.typ._convert_to_py(annotation.get(i))
      File "/Users/jigold/hail/python/hail/type.py", line 243, in _convert_to_py
        lst = env.jutils.iterableToArrayList(annotation)
      File "/Users/jigold/spark-2.0.2-bin-hadoop2.7/python/lib/py4j-0.10.3-src.zip/py4j/java_gateway.py", line 1133, in __call__
        answer, self.gateway_client, self.target_id, self.name)
      File "/Users/jigold/spark-2.0.2-bin-hadoop2.7/python/pyspark/sql/utils.py", line 63, in deco
        return f(*a, **kw)
      File "/Users/jigold/spark-2.0.2-bin-hadoop2.7/python/lib/py4j-0.10.3-src.zip/py4j/protocol.py", line 323, in get_return_value
        format(target_id, ".", name, value))
    Py4JError: An error occurred while calling o155.iterableToArrayList. Trace:
    py4j.Py4JException: Method iterableToArrayList([class [D]) does not exist
        at py4j.reflection.ReflectionEngine.getMethod(ReflectionEngine.java:318)
        at py4j.reflection.ReflectionEngine.getMethod(ReflectionEngine.java:326)
        at py4j.Gateway.invoke(Gateway.java:272)
        at py4j.commands.AbstractCommand.invokeMethod(AbstractCommand.java:132)
        at py4j.commands.CallCommand.execute(CallCommand.java:79)
        at py4j.GatewayConnection.run(GatewayConnection.java:214)
        at java.lang.Thread.run(Thread.java:745)
```

**Comment by johnc1231** [comment]
Worked on this a little bit, seems to me that specifically lmmreg creates a global annotation that doesn't agree with signature. Pretty sure I've fixed it, but I'd like to ask one of you guys about it first.

**Comment by jbloom22** [comment]
Fixed by #1369
